core/primitives/src/offchain.rs

```diff
@@ -241,6 +276,12 @@ pub trait Externalities {
 	/// The transaction will end up in the pool and be propagated to others.
 	fn submit_transaction(&mut self, extrinsic: Vec<u8>) -> Result<(), ()>;
 
+	/// Returns information about the local node's network state.
+	fn network_state(&self) -> Result<OpaqueNetworkState, ()>;
+
+	/// Returns the locally configured authority public key, if available.
```

Comment:
What is the "authority public key"? There are several keys in use for authorities now.

Reply:
There exists [a trait](https://github.com/paritytech/substrate/blob/master/core/offchain/src/lib.rs#L61-L65) for that with [this implementation](https://github.com/paritytech/substrate/blob/master/core/service/src/lib.rs#L885-L908), which I call to get the current authority public key. This whole part is currently getting reworked in https://github.com/paritytech/substrate/pull/3034, so I thought it's fine in master?

Reply:
ok

Reply:
I started refactoring the offchain workers api (discussed with @tomusdrw) while @tomaka is refactoring the service factory/components. I'm on vacation till next week. The plan is to rebase #3034 on those PRs